src/dsp_tools/commands/validate_data/make_data_rdf.py

```diff
@@ -22,208 +27,147 @@
-        case BooleanValueDeserialised():
-            return _transform_into_bool(val, res_iri)
-        case DecimalValueDeserialised():
-            return _transform_decimal_value(val, res_iri)
+            return _make_one_value_with_xsd_data_type(
+                val=val,
+                res_iri=res_iri,
+                prop_type_info=RDF_LITERAL_PROP_TYPE_MAPPER[type(val)],
+            )
         case LinkValueDeserialised():
             return _transform_link_value(val, res_iri)
         case ListValueDeserialised():
             return _transform_list_value(val, res_iri)
-        case TimeValueDeserialised():
-            return _transform_time_value(val, res_iri)
-        case UriValueDeserialised():
-            return _transform_uri_value(val, res_iri)
         case _:
             raise InternalError(f"Unknown Value Type: {type(val)}")
 
 
-def _transform_into_xsd_string(
-    val: ValueDeserialised, res_iri: URIRef, func: Callable[[URIRef, Literal, URIRef], ValueRDF]
-) -> ValueRDF:
-    new_str = val.object_value if val.object_value is not None else ""
-    return func(URIRef(val.prop_name), Literal(new_str, datatype=XSD.string), res_iri)
-
-
-def _transform_into_xsd_integer(
-    val: ValueDeserialised, res_iri: URIRef, func: Callable[[URIRef, Literal, URIRef], ValueRDF]
-) -> ValueRDF:
-    content = (
-        Literal(val.object_value, datatype=XSD.integer)
-        if val.object_value is not None
-        else Literal("", datatype=XSD.string)
-    )
-    return func(URIRef(val.prop_name), content, res_iri)
-
-
-def _transform_into_bool(val: ValueDeserialised, res_iri: URIRef) -> ValueRDF:
-    match val.object_value:
-        case "1" | "true":
-            content = Literal(True, datatype=XSD.boolean)
-        case "0" | "false":
-            content = Literal(False, datatype=XSD.boolean)
-        case _:
-            content = Literal("", datatype=XSD.string)
-    return BooleanValueRDF(URIRef(val.prop_name), content, res_iri)
-
-
-def _transform_decimal_value(val: ValueDeserialised, res_iri: URIRef) -> ValueRDF:
-    content = (
-        Literal(val.object_value, datatype=XSD.decimal)
-        if val.object_value is not None
-        else Literal("", datatype=XSD.string)
-    )
-    return DecimalValueRDF(URIRef(val.prop_name), content, res_iri)
-
-
-def _transform_link_value(val: ValueDeserialised, res_iri: URIRef) -> ValueRDF:
-    content = val.object_value if val.object_value is not None else ""
-    return LinkValueRDF(URIRef(val.prop_name), DATA[content], res_iri)
-
-
-def _transform_list_value(val: ListValueDeserialised, res_iri: URIRef) -> ValueRDF:
+def _make_one_value_with_xsd_data_type(
+    val: ValueDeserialised, res_iri: URIRef, prop_type_info: RDFPropTypeInfo
+) -> Graph:
+    g = Graph()
+    val_iri = DATA[str(uuid4())]
+    g.add((val_iri, RDF.type, prop_type_info.knora_type))
+    if val.object_value:
+        literal_value = Literal(val.object_value, datatype=prop_type_info.xsd_type)
```

Comment:
The xsd ensures that integers and decimals are in fact numbers, therefore we only need to check that the value is not `None`. Interestingly `rdflib` accepts 0 and 1 as booleans and converts it.